prov/verbs/src/verbs_eq.c

```diff
@@ -884,6 +981,16 @@ static int fi_ibv_eq_close(fid_t fid)
 
 	dlistfd_head_free(&eq->list_head);
 
+	/* Detach all domains associated to the given EQ. No lock is needed since
+	   polling an EQ after it gets closed is prohibited. */
+	dlist_foreach_container_safe(&eq->domain_list, struct fi_ibv_domain,
+	                             domain, eq_dlentry, tmp_entry) {
+		/* The EQ that is closing should match the one in the domain. */
+		assert(domain->eq == eq);
+		fi_ibv_eq_dettach_domain(domain->eq, domain);
+		domain->eq = NULL;
+	}
```

Comment:
This is racy if the app tries to close the EQ and domain from different threads.

But we shouldn't need this.  Since the domain is bound to the EQ, the domain must be destroyed first.  Reference counting should provide the needed protection here.

Reply:
Thanks, I've added a reference counter for the EP.